src/apps/search-header/search-header.tsx

```diff
@@ -80,29 +96,66 @@ const SearchHeader: React.FC<SearchHeaderProps> = ({
     setCurrentlySelectedItem(determinSuggestionType(selectedItem));
   }
 
+  // downshift prevents the default form submission event when the autosuggest
+  // is open - that's why in some cases we have to simulate form sumbission
+  function manualRedirect(inputValue: string) {
+    const baseUrl = t("searchHeaderUrlText");
+    const params = inputValue;
+    if (window.top) {
+      window.top.location.href = `${baseUrl}?q=${params}`;
+    }
+  }
+
   function handleHighlightedIndexChange(
     changes: UseComboboxStateChange<Suggestion>
   ) {
+    const { selectedItem, highlightedIndex, type } = changes;
+    if (!selectedItem || !highlightedIndex) {
+      return;
+    }
     if (
-      changes.selectedItem === undefined ||
-      changes.selectedItem === null ||
-      changes.highlightedIndex === undefined
+      type === downshiftEventTypes.item_mouse_move ||
+      type === downshiftEventTypes.menu_mouse_leave
     ) {
       return;
     }
-    if (changes.highlightedIndex > -1) {
-      const arrayIndex: number = changes.highlightedIndex;
-      const currentlyHighlightedObject = suggestItems[arrayIndex];
-      const currentItemValue = determinSuggestionType(
-        currentlyHighlightedObject
-      );
-      setQ(currentItemValue);
-    } else {
+    if (highlightedIndex < 0) {
       setIsAutosuggestOpen(false);
+      return;
     }
+    const arrayIndex: number = highlightedIndex;
+    const currentlyHighlightedObject = orderedData[arrayIndex];
+    const currentItemValue = determinSuggestionType(currentlyHighlightedObject);
+    if (type === downshiftEventTypes.controlled_prop_updated_selected_item) {
+      manualRedirect(currentItemValue);
+    }
+    if (
+      type === downshiftEventTypes.input_keydown_arrow_down ||
+      type === downshiftEventTypes.input_keydown_arrow_up
```

Comment:
Here you can also use this construct:
```ts
[
  downshiftEventTypes.input_keydown_arrow_down,
  downshiftEventTypes.input_keydown_arrow_up
].includes(type)

Reply:
Do you think that's cleaner? What about consistency then?